prefect_github/filesystem.py

```diff
@@ -0,0 +1,149 @@
+    _logo_url = "https://images.ctfassets.net/gm98wzqotmnx/187oCWsD18m5yooahq1vU0/ace41e99ab6dc40c53e5584365a33821/github.png?h=250"  # noqa: E501
+
+    repository: str = Field(
+        default=...,
+        description=(
+            "The URL of a GitHub repository to read from, in either HTTPS or SSH "
+            "format. If you are using a private repo, it must be in the HTTPS format."
+        ),
+    )
+    reference: Optional[str] = Field(
+        default=None,
+        description="An optional reference to pin to; can be a branch name or tag.",
+    )
+    credentials: Optional[GitHubCredentials] = Field(
+        default=None,
+        description="An optional GitHubCredentials block for using private GitHub repos.",  # noqa: E501
+    )
+
+    @validator("credentials")
+    def _ensure_credentials_go_with_https(cls, v: str, values: dict):
+        """Ensure that credentials are not provided with 'SSH' formatted GitHub URLs."""
+        if v is not None:
+            if not values["repository"].startswith("https://"):
+                raise InvalidRepositoryURLError(
+                    (
+                        "Crendentials can only be used with GitHub repositories "
+                        "using the 'HTTPS' format. You must either remove the "
+                        "credential if you wish to use the 'SSH' format and are not "
+                        "using a private repository, or you must change the repository "
+                        "url to the 'HTTPS' format. "
+                    )
+                )
+
+        return v
+
+    def _create_repo_url(self) -> str:
+        """Format the URL provided to the `git clone` command.
+
+        For private repos: https://<oauth-key>@github.com/<username>/<repo>.git
+        All other repos should be the same as `self.repository`.
+        """
+
+        if self.repository.startswith("https://") and self.credentials is not None:
+            repo_url = self.repository[8:]
+            token_value = self.credentials.token.get_secret_value()
+            full_url = f"https://{token_value}@{repo_url}"
+        else:
+            full_url = self.repository
+
+        return full_url
+
+    async def get_directory(
+        self, from_path: str = None, local_path: str = None
+    ) -> None:
+        """
+        Clones a GitHub project specified in `from_path` to the provided `local_path`;
+        defaults to cloning the repository reference configured on the Block to the
+        present working directory.
+
+        Args:
+            from_path: If provided, interpreted as a subdirectory of the underlying
+                repository that will be copied to the provided local path.
+            local_path: A local path to clone to; defaults to present working directory.
+        """
+
+        # CONSTRUCT COMMAND
+        cmd = f"git clone {self._create_repo_url()}"
+        if self.reference:
+            cmd += f" -b {self.reference} --depth 1"
```

Comment:
Curious what the depth does?

Also maybe `-b` can be full name of the option, --branch (or the actual option)

Reply:
"Create a shallow clone with a history truncated to the specified number of commits". It keeps us from bringing the whole history with the clone command.